fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraVersioning.java

```diff
@@ -201,15 +201,15 @@ public Response addVersion(@HeaderParam(MEMENTO_DATETIME_HEADER) final String da
                                 UNSUPPORTED_MEDIA_TYPE);
                     }
 
-                    final FedoraResource rdfMemento = versionService.createVersion(session.getTransaction(),
+                    final FedoraResource rdfMemento = versionService.createVersion(transaction,
                             originalResource, idTranslator, mementoInstant, bodyStream, format);
                     // If a binary memento was also generated, use the binary in the response
                     if (!isBinary) {
                         memento = rdfMemento;
                     }
                 }
 
-                session.commit();
+                transaction.commit();
```

Comment:
maybe it would be helpful to have a `commitIfShortLived` method since this is a very common case

Reply:
+1

Reply:
Done